term_test.go

```diff
@@ -1,42 +1,34 @@
-//+build linux
+// +build !windows
 
 package term
 
 import (
 	"io/ioutil"
 	"os"
-	"strings"
 	"testing"
 
+	"github.com/creack/pty"
 	"github.com/google/go-cmp/cmp"
 	"gotest.tools/assert"
 )
 
-// RequiresRoot skips tests that require root, unless the test.root flag has
-// been set
-func RequiresRoot(t *testing.T) {
-	if os.Getuid() != 0 {
-		t.Skip("skipping test that requires root")
-		return
+func newTtyForTest(t *testing.T) (*os.File, *os.File, error) {
+	pty, tty, err := pty.Open()
+	if err != nil {
+		t.Fatalf("error creating pty: %v", err)
 	}
-}
 
-func newTtyForTest(t *testing.T) (*os.File, error) {
-	RequiresRoot(t)
-	file, err := os.OpenFile("/dev/tty", os.O_RDWR, os.ModeDevice)
-	if err != nil && strings.Contains(err.Error(), "no such device or address") {
-		t.Skip("terminal missing, skipping test")
-	}
-	return file, err
+	return pty, tty, err
 }
 
 func newTempFile() (*os.File, error) {
 	return ioutil.TempFile(os.TempDir(), "temp")
 }
 
 func TestGetWinsize(t *testing.T) {
-	tty, err := newTtyForTest(t)
+	pty, tty, err := newTtyForTest(t)
 	assert.NilError(t, err)
```

Comment:
See comment above; this looks redundant (or duplicating) what's done in `newTtyForTest`